src/lib.rs

```diff
@@ -0,0 +1,108 @@
+
+extern crate time;
+extern crate url;
+
+use std::collections::HashMap;
+use time::Timespec;
+use url::Url;
+
+#[derive(Clone, Debug)]
+pub struct PlaceAction {
+    pub url: Url,
+    pub when: Timespec,
+    pub event: String,
+    // TODO: Think about more complex types -- Or do we want callers just
+    // storing JSON here? Can we use std::any::Any and hope to serialize it?
+    pub data: String
```

Comment:
Yeah, that was my thought too. And since Any exists already, it seems like the trait we might want to use if we were going to go the route of allowing anything.